dojo/tools/threat_composer/parser.py

```diff
@@ -0,0 +1,141 @@
+
+    def get_label_for_scan_types(self, scan_type):
+        return "ThreatComposer Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "ThreatComposer report file can be imported in JSON format."
+
+    def get_findings(self, file, test):
+        data = json.load(file)
+        findings = []
+
+        if "threats" not in data:
+            msg = "Invalid ThreatComposer data"
+            raise ValueError(msg)
+
+        if "assumptionLinks" in data:
+            assumptions = {assumption["id"]: assumption for assumption in data["assumptions"]}
+            assumption_mitigation_links = defaultdict(list)
+            assumption_threat_links = defaultdict(list)
+            for link in data["assumptionLinks"]:
+                linked_id = link["linkedId"]
+                assumption_id = link["assumptionId"]
+                assumption_type = link["type"]
+                if assumption_id in assumptions:
+                    if assumption_type == "Threat":
+                        assumption_threat_links[linked_id].append(assumptions[assumption_id])
+                    elif assumption_type == "Mitigation":
+                        assumption_mitigation_links[linked_id].append(assumptions[assumption_id])
+
+        if "mitigationLinks" in data:
+            mitigations = {
+                mitigation["id"]: {
+                    "mitigation": mitigation,
+                    "assumptions": assumption_mitigation_links[mitigation["id"]],
+                }
+                for mitigation in data["mitigations"]
+            }
+            mitigation_links = defaultdict(list)
+            for link in data["mitigationLinks"]:
+                linked_id = link["linkedId"]
+                mitigation_id = link["mitigationId"]
+                if mitigation_id in mitigations:
+                    mitigation_links[linked_id].append(mitigations[mitigation_id])
+
+        for threat in data["threats"]:
+            if threat["threatAction"]:
+                title = threat["threatAction"]
+                severity, impact, comments = self.parse_threat_metadata(threat["metadata"])
+                description = self.to_description_text(threat, comments, assumption_threat_links[threat["id"]])
+                mitigation = self.to_mitigation_text(mitigation_links[threat["id"]])
+                unique_id_from_tool = threat["id"]
+                vuln_id_from_tool = threat["numericId"]
+                tags = threat["tags"] if "tags" in threat else []
+
+                finding = Finding(
+                    title=title,
+                    description=description,
+                    severity=severity,
+                    vuln_id_from_tool=vuln_id_from_tool,
+                    unique_id_from_tool=unique_id_from_tool,
+                    mitigation=mitigation,
+                    impact=impact,
+                    tags=tags,
+                    static_finding=False,
+                    dynamic_finding=False,
```

Comment:
```suggestion
                    static_finding=True,
                    dynamic_finding=False,
```

While a threat modeling finding is a little different than a typical "static" or "dynamic" finding produced by e.g. SAST/DAST scans, I think "static" probably makes the most sense of the two options available. By setting both of these to `False`, the finding will default to "dynamic".

![Screenshot 2024-09-05 at 14 58 40](https://github.com/user-attachments/assets/a758086a-645f-4e06-838e-557e5fd5c419)

Reply:
changed